pytorch_lightning/strategies/ddp.py

```diff
@@ -342,20 +342,20 @@ def training_step(self, *args, **kwargs) -> STEP_OUTPUT:
 
     def validation_step(self, *args, **kwargs) -> Optional[STEP_OUTPUT]:
         with self.precision_plugin.val_step_context():
-            if isinstance(self.model, DistributedDataParallel):
+            if self.lightning_module.trainer.state.fn == TrainerFn.FITTING:
```

Comment:
How does this impact fsdp? fairscale fsdp is still inheritance from ddp right?

Reply:
yes, but these methods are overridden there.